### PR TITLE
Eip4844 containers

### LINKED ISSUE #6383: Add\update containers
### New SSZ containers

[p2p containers](https://github.com/ethereum/consensus-specs/blob/dev/specs/eip4844/p2p-interface.md#containers)

- [ ] `BlobsSidecar`
- [ ] `SignedBeaconBlockAndBlobsSidecar`

### Updated SSZ containers
[beacon containers](https://github.com/ethereum/consensus-specs/blob/dev/specs/eip4844/beacon-chain.md#containers)
- [ ] `BeaconBlockBody`
  - [ ] `BeaconBlockBodyBuilder`
- [ ] `ExecutionPayload`
- [ ] `ExecutionPayloadHeader`

### PR DESCRIPTION
**Pull request by zilm13**
## PR Description
- current spec's `ExecutionPayload`, `ExecutionPayloadHeader`, `BeaconBlockBody` are built on top of the Bellatrix, I've built it on top of the Capella keeping Capella's vars. I think we expect that all spec will be updated to be on top of the Capella, but there are already running testnets for EIP-4844 which are built on top of the Bellatrix. If we want to participate in it, it's a wrong way to go, but we already are building fork on top of the Capella. So my uncertainty comes... What do you think? 
- do we really want this here? https://github.com/ConsenSys/teku/blob/e2df22b6935a30478362b875fc18970a3552c4e8/ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayload.java#L31-L33 
refactored it a bit in https://github.com/ConsenSys/teku/commit/adc283c5a8f1503deffa588f497078ddb9d451a5 to not leak with Capella child properties. I'm ok with `ExecutionPayloadFields`, enum is flawed on arrival, but we could protect base interfaces at least.
- same thing in https://github.com/ConsenSys/teku/blob/d4b2b09a71975c19cd5c12979f32e07a13e48259/ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/BeaconBlockBodyBuilder.java#L50-L63 Doesn't hurt me that much as it's support class plus `Supplier`, anyway, I could either follow this pattern or try to refactor
- tests needed, but all requires `SpecLogic`: `BlobsSidecar` ssz roundtrip, `SignedBeaconBlockAndBlobsSidecar` ssz rountrip, `ExecutionPayload` ssz rountrip,  `ExecutionPayloadHeader` ssz roundtrip, `BeaconBlockBody` ssz rountrip, `BlindedBeaconBlockBody` ssz roundtrip, general `BeaconBlockBody` and `ExecutionPayload` tests on EIP-4844 `Milestone`
- serializable data structures in `tech.pegasys.teku.api.schema` requires `SchemaDefinition`, postponed

## Fixed Issue(s)

Fixes #6383 
## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/versions/eip4844/ExecutionPayloadHeaderSchemaEip4844.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    return executionPayloadHeaderOfDefaultPayload;
+  }
+
+  @Override
+  public ExecutionPayloadHeaderEip4844Impl createFromBackingNode(final TreeNode node) {
+    return new ExecutionPayloadHeaderEip4844Impl(this, node);
+  }
+
+  @Override
+  public ExecutionPayloadHeaderEip4844Impl createFromExecutionPayload(
```
[the remark this answers is not in the document]
Comment (zilm13) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/versions/eip4844/ExecutionPayloadHeaderSchemaEip4844.java:168]: Fixed with https://github.com/ConsenSys/teku/pull/6434/commits/4aae04786663b3d384b167d735adff7285e438a7

**Comment by tbenr** [comment]
> do we really want this here?
[teku/ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayload.java](https://github.com/ConsenSys/teku/blob/e2df22b6935a30478362b875fc18970a3552c4e8/ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayload.java#L31-L33)

not sure about this change... if you see `tech.pegasys.teku.spec.datastructures.blocks.blockbody.BeaconBlockBody` we have similar optionals for the interface. It give us a "feature oriented" approach in addition to the strictly "version oriended" approach.

**Comment by zilm13** [comment]
Ok we are going in consistent way with `ExecutionPayload` and `BeaconBlockBodyBuilder`, pushed changes

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/eip4844/BeaconBlockBodySchemaEip4844.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+package tech.pegasys.teku.spec.datastructures.blocks.blockbody.versions.eip4844;
+
+import java.util.Optional;
+import tech.pegasys.teku.infrastructure.ssz.schema.SszListSchema;
+import tech.pegasys.teku.spec.datastructures.blocks.blockbody.versions.capella.BeaconBlockBodySchemaCapella;
+import tech.pegasys.teku.spec.datastructures.type.SszKZGCommitment;
+
+public interface BeaconBlockBodySchemaEip4844<T extends BeaconBlockBodyEip4844>
+    extends BeaconBlockBodySchemaCapella<T> {
+
```
Comment (tbenr) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/eip4844/BeaconBlockBodySchemaEip4844.java:26]: Like for the others, we probably need the `required` method here

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/eip4844/BlindedBeaconBlockBodySchemaEip4844.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+package tech.pegasys.teku.spec.datastructures.blocks.blockbody.versions.eip4844;
+
+import java.util.Optional;
+import tech.pegasys.teku.infrastructure.ssz.schema.SszListSchema;
+import tech.pegasys.teku.spec.datastructures.blocks.blockbody.versions.capella.BlindedBeaconBlockBodySchemaCapella;
+import tech.pegasys.teku.spec.datastructures.type.SszKZGCommitment;
+
+public interface BlindedBeaconBlockBodySchemaEip4844<T extends BlindedBeaconBlockBodyEip4844>
+    extends BlindedBeaconBlockBodySchemaCapella<T> {
+
```
Comment (tbenr) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/eip4844/BlindedBeaconBlockBodySchemaEip4844.java:26]: here we are missing a bunch  of `required` from the previous forks

**Review by tbenr — APPROVED** [review]
LGTM. just a couple of nit about `required` method

**Pull request merged**